### PR TITLE
New unpooling, variable placeholders dimension, trimap argument for test script.

### PR DESCRIPTION
**Pull request by sashulyak**
1. Fresh version of unpooling added. Now it can deal with variable batch size. (took it from here https://github.com/tensorflow/tensorflow/issues/2169)
2. 0-dimension of placeholders were set to None. This allows setting a different batch size for train and test runs.
3. Trimap argument added to the test script, so now you can test pretrained model on your own images, even if you don't have alpha-mask.
4. Redme edited.

### CODE REVIEW

**Comment by ImMrMa** [comment]
Thank you very much for your help with this image size, but I see that after you change the code, the input image will still be resized to that 320 pixel , so how can you make it has the size of according to the input size?
